### PR TITLE
Subscription Upgrades

### PR DESCRIPTION
**Pull request by braxtonhall**
- Show box name if the screen is big enough
- Boxes can have NO OWNER!!!
- Added graceful shutdowns -- finishes all currently running jobs before exiting
- Set a max concurrency on the jobs because prisma is broken
- Boxes made for secret dj are maintained by secret dj users, not owned
	- this means that they can still archive/manage comments, but they cannot change the stylesheet, origin, or box name
- Reply emails are now only sent once the message can no longer be deleted
- You can unsubscribe from ALL emails from the /subscriptions page
- The "subscribe to this box" is auto filled if you are logged in (not really going to get used ever)

### CODE REVIEW

File: src/secret-dj/operations/createGame.ts [hunk 1]
```diff
@@ -27,35 +27,34 @@ export const createGame = async ({
...
+					ownerId,
+					description,
+					boxId: box.id,
+					unlisted,
+				},
+				select: {
+					id: true,
+				},
+			}),
+		]);
```
Comment (aymendirar) [inline src/secret-dj/operations/createGame.ts:57]: lmao is this pattern what prompted the max jobs stuff

File: src/server.ts [hunk 2]
```diff
@@ -11,20 +11,27 @@ import { api as unauthenticatedApi, views as unauthenticatedViews } from "./tool
...
-import { checkboxSchema } from "./util/checkboxSchema";
+import { settingsSchema } from "./schema";
 
 // TODO would be great to also serve a javascript client
 // TODO this whole system is a mess...
 
 const views = express()
 	.set("view engine", "ejs")
 	.use("/public", express.static("public"))
+	.use((req, res, next) => {
```
Comment (aymendirar) [inline src/server.ts]: nit: make this into middleware function that describes what this is doing/why

File: src/toolbox/operations/informParents.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	});
+	await enqueue({
+		persona: EmailPersona.BOBS_MAILER,
+		address: env.address,
+		subject: "You received a new reply",
+		html,
+	});
+};
+
+export const informParents = () =>
```
Comment (aymendirar) [inline src/toolbox/operations/informParents.ts:36]: neattt

File: src/toolbox/operations/informParents.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+						},
+					},
+				},
+			},
+		});
+		await AsyncPool.mapToSettled(
+			replies,
+			async ({ boxId, id, parent, content, emailId }) => {
+				if (
+					parent &&
```
Comment (aymendirar) [inline src/toolbox/operations/informParents.ts:79]: is the email notif only for replies to messages in a comment box? do i get notified if any non-reply message is posted to a comment box that i maintain?

**Review by aymendirar — APPROVED** [review]
epic

File: src/toolbox/operations/informParents.ts (hunk 4 above)
Comment (braxtonhall) [reply src/toolbox/operations/informParents.ts:79]: Yeah that happens in a different job. it's called updateSubscribers or something like that

File: src/server.ts (hunk 2 above)
Comment (braxtonhall) [reply src/server.ts]: yeah..........

File: src/secret-dj/operations/createGame.ts (hunk 1 above)
Comment (braxtonhall) [reply src/secret-dj/operations/createGame.ts:57]: lol no this here is fine. it's just i am afraid of a LOT of replies coming in all at once

**Pull request merged**